backend/src/v1/complaint/complaint.service.ts

```diff
@@ -1008,13 +1058,30 @@ export class ComplaintService {
         });
       }
 
-
       //-- added this for consistency with search method
       //-- return Waste and Pestivide complaints for CEEB users
       if (hasCEEBRole && complaintType === "ERS") {
         unMappedBuilder.andWhere("violation_code.agency_code = :agency", { agency: "EPO" });
       }
 
+      // -- filter by complaint identifiers returned by case management if actionTaken filter is present
+      if (hasCEEBRole && filters.actionTaken) {
+        const { data, errors } = await get(token, {
+          query: `{getLeadsByActionTaken (actionCode: "${filters.actionTaken}")}`,
+        });
+        if (errors) {
+          this.logger.error("GraphQL errors:", errors);
+          throw new Error("GraphQL errors occurred");
+        }
+        // If no complaint indentifiers are returned by the CM database, provide a non-empty non-matching list to avoid
+        // the SQL error of IN empty set.
+        const complaintIdentifiers = data.getLeadsByActionTaken.length > 0 ? data.getLeadsByActionTaken : [-1];
```

Comment:
Sorry forgot a comment - looks like we are over our complexity warning for this method.   

Could this be encapsulated into a function and not repeated from earlier (lines: 1006-1031)

Reply:
Arg, we're so close... also it looks like there is one extra item that's been introduced with this change.   See 

https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=708&id=bcgov_nr-compliance-enforcement&open=AZKWOerdTDM-Jdj3eJdr

Not sure if you have another idea for how to simplify that a bit?